dandi/files/bids.py

```diff
@@ -174,6 +200,10 @@ def get_metadata(
         metadata["path"] = self.path
         return BareAsset(**metadata)
 
+    def get_validation_bids_version(self) -> str:
+        assert self.bids_dataset_description._bids_version
```

Comment:
Shouldn't there be a call to `self._validate()` before this?

Reply:
@jwodder you're right. Fixed in [1e2d6e7](https://github.com/dandi/dandi-cli/pull/1164/commits/1e2d6e72ac3e9f6b2c3244702ea5bdad6c0beb0a). Does that also mean the assert line can be removed?

Reply:
Actually, that should be `self.bids_dataset_description._validate()`, and you still need the `assert` to make mypy happy.